lib/runtime/src/ops/layer_norm.cc

```diff
@@ -16,566 +16,197 @@
-         _input->data_type,
-         name,
-         1 /*inputs*/,
-         _elementwise_affine ? 2 : 0 /*weights*/,
-         1 /*outputs*/,
-         _input),
-      elementwise_affine(_elementwise_affine), eps(_eps), axes(_axes) {
-  // overwrite layer_guid
-  layer_guid = _layer_guid;
-  outputs[0] = model.create_parallel_tensor_legion_ordering(
-      _input->num_dims, _input->dims, _input->data_type, this);
-  assert(check_output_input_weight_parallel_dims(allocate_weights));
-  ParallelDim output_dims[MAX_TENSOR_DIM];
-  int M = 1;
-  for (int i = 0; i < axes.size(); i++) {
-    M *= inputs[0]->dims[inputs[0]->num_dims - 1 - axes[i]].size;
-  }
-  effective_num_elements = M;
-  effective_batch_size = inputs[0]->get_volume() / M;
-  if (numWeights > 0 && allocate_weights) {
-    int kernel_dims = 2;
-    assert(false);
-    // weights[0] = model.create_parallel_weight_legion_ordering(
-    //     kernel_dims,
-  } else {
-    // do nothing
-  }
-  return;
+OpTaskInvocation backward(LayerNormAttrs const &attrs) {
+  OpTaskBinding b = infer_bwd_binding(forward(attrs).binding);
+
+  return {LAYERNORM_BWD_TASK_ID, b};
 }
 
-void LayerNorm::init(FFModel const &ff) {
-  assert(check_output_input_weight_same_parallel_is());
-  parallel_is = outputs[0]->parallel_is;
-  ArgumentMap argmap;
-  Context ctx = ff.config.lg_ctx;
-  Runtime *runtime = ff.config.lg_hlr;
-  set_argumentmap_for_init(ff, argmap);
-  IndexLauncher launcher(LAYERNORM_INIT_TASK_ID,
-                         parallel_is,
-                         TaskArgument(this, sizeof(LayerNorm)),
-                         argmap,
-                         Predicate::TRUE_PRED,
-                         false /*must*/,
-                         0 /*mapper_id*/,
-                         outputs[0]->machine_view.hash());
-  launcher.add_region_requirement(RegionRequirement(outputs[0]->part,
-                                                    0 /*projection id*/,
-                                                    WRITE_ONLY,
-                                                    EXCLUSIVE,
-                                                    outputs[0]->region));
-  launcher.add_field(0, FID_DATA);
-  launcher.add_region_requirement(RegionRequirement(inputs[0]->part,
-                                                    0 /*projection id*/,
-                                                    READ_ONLY,
-                                                    EXCLUSIVE,
-                                                    inputs[0]->region));
-  launcher.add_field(1, FID_DATA);
-  FutureMap fm = runtime->execute_index_space(ctx, launcher);
-  fm.wait_all_results();
-  set_opmeta_from_futuremap(ff, fm);
+static optional<float> forward_task_impl(TaskArgumentAccessor const &acc) {
+  auto input = acc.get_tensor<Permission::RO>(INPUT);
+  auto output = acc.get_tensor<Permission::WO>(OUTPUT);
+  auto gamma = acc.get_tensor<Permission::WO>(GAMMA);
+  auto beta = acc.get_tensor<Permission::WO>(BETA);
```

Comment:
```suggestion
  auto gamma = acc.get_tensor<Permission::RW>(GAMMA);
  auto beta = acc.get_tensor<Permission::RW>(BETA);
```

Reply:
why "RW"

Reply:
Based on original permissions, I think it's RW and for backward only the grads are RW